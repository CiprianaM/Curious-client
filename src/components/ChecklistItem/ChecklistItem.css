.checklist__item>div {
  display: flex;
  width: 50%;
  align-items: center;
  justify-content: flex-start;
}

.checklist__item>div>* {
  font-family: 'Comfortaa';
}