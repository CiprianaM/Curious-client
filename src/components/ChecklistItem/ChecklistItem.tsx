import React, { useState } from 'react';
import PropTypes from 'prop-types';

interface IchecklistItem {
  id: string
  title: string
  completed: boolean
}

interface ChecklistItemProps {
  checklistItem: IchecklistItem
  checklistTitleInput: string
  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  handleChecked: (e: React.ChangeEvent<HTMLInputElement>) => void
  handleDeleteChecklistItem: (checklistItemId: string) => void
  handleUpdateChecklistItem: (checklistItemId: string) => void
}

const ChecklistItem: React.FC<ChecklistItemProps> = ({
  checklistItem,
  checklistTitleInput,
  handleChange,
  handleChecked,
  handleDeleteChecklistItem,
  handleUpdateChecklistItem,
}) => {
  const [editing, setEditing] = useState(false);

  const handleEditing = () => {
    setEditing(!editing);
  };

  const handleUpdateButton = () => {
    handleUpdateChecklistItem(checklistItem.id);
    handleEditing();
  };

  const handleDeleteButton = () => {
    handleDeleteChecklistItem(checklistItem.id);
    handleEditing();
  };

  return (
    <div>
      {editing
        ? (
          <div>
            <input type="text" value={checklistTitleInput} onChange={handleChange} />
            <button type="button" onClick={() => handleUpdateButton()}>Save Changes</button>
            <button type="button" onClick={() => handleDeleteButton()}><span>x</span></button>
          </div>
        )
        : (
          <div>
            <input type="checkbox" id={checklistItem.id} checked={checklistItem.completed} onChange={handleChecked} />
            <label htmlFor={checklistItem.id} onClick={handleEditing}>{checklistItem.title}</label>
          </div>
        )}
    </div>
  );
};

ChecklistItem.propTypes = {
  checklistItem: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    completed: PropTypes.bool.isRequired,
  }).isRequired,
  checklistTitleInput: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleChecked: PropTypes.func.isRequired,
  handleDeleteChecklistItem: PropTypes.func.isRequired,
  handleUpdateChecklistItem: PropTypes.func.isRequired,

};
export default ChecklistItem;
