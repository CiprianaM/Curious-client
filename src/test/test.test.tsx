import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

import {
  configure, shallow, mount, render,
} from 'enzyme';


import ChecklistItem from '../components/ChecklistItem/ChecklistItem';

configure({ adapter: new Adapter() });


describe('My first test suite!', () => {
  const fn = () => '';
  it('correctly renders', () => {
    const wrapper = shallow(<ChecklistItem checklistItem={{ id: '1', completed: false, title: 'hello' }} handleChecked={fn} onChange={fn} handleDeleteChecklistItem={fn} handleUpdateChecklistItem={fn} />);
    expect(wrapper).toMatchSnapshot();
  });
});
