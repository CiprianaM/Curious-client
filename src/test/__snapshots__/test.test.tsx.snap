// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first test suite! correctly renders 1`] = `ShallowWrapper {}`;
